File: tmp/sys/SwordOffer/src/main/java/com/rann/offer/problem7/MyQueue.java
package com.rann.offer.problem7;

import java.util.Stack;

/**
 * Problem7
 * 两个栈实现队列
 *
 * @author lemonjing
 * @version 1.0
 */
public class MyQueue {

    private Stack<Integer> stack1 = new Stack<>();
    private Stack<Integer> stack2 = new Stack<>();

    public void appendTail(Integer a) {
        stack1.push(a);
    }

    public Integer deleteHead() throws Exception {
        if (stack2.isEmpty()) {
            while (!stack1.isEmpty()) {
                stack2.push(stack1.pop());
            }
        }

        if (stack2.isEmpty()) {
            throw new Exception("队列为空，不能删除");
        }

        return stack2.pop();
    }
}

File: tmp/sys/SwordOffer/src/main/scala/com/scala/offer/Problem7/MyQueue.scala
package com.scala.offer.Problem7

import java.util

/**
 * Problem7
 * 两个栈实现队列
 *
 * @author lemonjing
 */

object MyQueue {

  private val stack1 = new util.Stack[Integer]()
  private val stack2 = new util.Stack[Integer]()

  def push(e: Integer): Unit = {
    stack1.push(e)
  }

  def pop(): Integer = {
    if (stack2.isEmpty) {
      while (!stack1.isEmpty) {
        stack2.push(stack1.pop())
      }
    }
    if (stack2.isEmpty) {
      throw new Exception("delete error, queue is empty.")
    }
    stack2.pop()
  }
}
